snapcraft/application.py

```diff
@@ -41,6 +43,11 @@
 class Snapcraft(Application):
     """Snapcraft application definition."""
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Whether we know that we should use the core24-based codepath.
+        self._known_core24 = False
```

Comment:
Maybe `self._known_core = "core24"`? So we can handle multiple core fallback.

Reply:
Are you thinking of future `core26` cases or previous `core22` and `core20`?

Reply:
I was thinking the `core22` and `core20`

Reply:
On in this case the current code already handles this - this code checks that the base isn't `core24`, then raises the `ClassicFallback` error. The non-craft-app-based code then has [its own handling](https://github.com/snapcore/snapcraft/blob/98550856e8516d956fa32101472ceb1d0dedd025/snapcraft/cli.py#L263) if the base isn't `core22`. So there's no need to handle multiple types of fallbacks here.